akka-persistence/src/main/scala/akka/persistence/journal/inmem/InmemJournal.scala

```diff
@@ -87,16 +87,16 @@ private[persistence] class InmemJournal(cfg: Config) extends AsyncWriteJournal w
 private[persistence] trait InmemMessages {
   // persistenceId -> persistent message
   var messages = Map.empty[String, Vector[PersistentRepr]]
+  // persistenceId -> highest used sequence number
+  private var highestSequenceNumbers = Map.empty[String, Long]
```

Comment:
Makes me wonder if we'd need a faster data structure or if this isn't quite on the hot path then no.

Reply:
inmem journal is only intended for testing so shouldn't be a problem